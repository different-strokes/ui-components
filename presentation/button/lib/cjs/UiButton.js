'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var UiButton = function UiButton() {
  return _react2.default.createElement(
    'div',
    null,
    _react2.default.createElement(
      'p',
      null,
      'This is UiButton'
    )
  );
};

UiButton.propTypes = {};

UiButton.defaultProps = {};

exports.default = UiButton;
//# sourceMappingURL=UiButton.js.map