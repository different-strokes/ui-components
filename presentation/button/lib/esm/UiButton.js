import React from 'react';

var UiButton = function UiButton() {
  return React.createElement(
    'div',
    null,
    React.createElement(
      'p',
      null,
      'This is UiButton'
    )
  );
};

UiButton.propTypes = {};

UiButton.defaultProps = {};

export default UiButton;
//# sourceMappingURL=UiButton.js.map