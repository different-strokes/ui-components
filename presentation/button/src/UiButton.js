import React from 'react';
import PropTypes from 'prop-types';

const UiButton = ({ onClick, text, theme }) => (
  <button onClick={onClick} className={theme.button}>
    {text}
  </button>
);

UiButton.propTypes = {
  onClick: PropTypes.func,
  text: PropTypes.oneOf([
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
  theme: PropTypes.shape({
    button: PropTypes.string,
  }),
};

UiButton.defaultProps = {
  onClick: function noop() {},
  theme: {},
};

export default UiButton;
