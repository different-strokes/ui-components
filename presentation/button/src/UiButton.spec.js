import React from 'react';
import renderer from 'react-test-renderer';

import UiButton from './UiButton';

describe('UiButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <UiButton />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
