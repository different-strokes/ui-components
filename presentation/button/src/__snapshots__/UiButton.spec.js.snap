// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiButton renders correctly 1`] = `
<div>
  <p>
    This is UiButton
  </p>
</div>
`;
